src/platforms/Token/TokenRewardContext.tsx

```diff
@@ -0,0 +1,47 @@
+import useTokenData from "hooks/useTokenData"
+import { ReactElement, createContext, useContext } from "react"
+import { GuildPlatform } from "types"
+import useTokenClaimFee from "./hooks/useClaimToken"
+
+export type TokenRewardContextType = {
+  guildPlatform: GuildPlatform
+  token: { data: ReturnType<typeof useTokenData>["data"]; isLoading: boolean }
+  fee: { amount: bigint; isLoading: boolean }
+  imageUrl: string
+}
+
+const TokenRewardProvider = ({
+  children,
+  guildPlatform,
+}: {
+  children: ReactElement
+  guildPlatform: GuildPlatform
+}): JSX.Element => {
+  const {
+    platformGuildData: { tokenAddress, chain, imageUrl },
+  } = guildPlatform || { platformGuildData: {} }
+  const { amount, isLoading: isFeeLoading } = useTokenClaimFee(chain)
+  const { data: token, isLoading: isTokenLoading } = useTokenData(
```

Comment:
The same could be true about the useTokenClaimFee hook. I am not sure about these, it could make our lives easier to fetch these often use thing in the context instead of having to do this separately every time. I was thinking of moving the claimable amount calculation here as well.